pkg/gitops/init_test.go

```diff
@@ -0,0 +1,440 @@
+					{
+						Ports: []crdv1alpha1.NetworkPolicyPort{
+							{
+								Port: &int81,
+							},
+						},
+						To: []crdv1alpha1.NetworkPolicyPeer{
+							{
+								PodSelector:       &selectorB,
+								NamespaceSelector: &selectorC,
+							},
+						},
+						Action: &allowAction,
+					},
+				},
+			},
+		},
+		expPath: "/antrea-cluster-policies/cnpA.yaml",
+		expYaml: `apiVersion: crd.antrea.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: cnpA
+spec:
+  appliedTo:
+  - podSelector:
+      matchLabels:
+        foo1: bar1
+  egress:
+  - action: Allow
+    enableLogging: false
+    from: null
+    name: ""
+    ports:
+    - port: 81
+    to:
+    - namespaceSelector:
+        matchLabels:
+          foo3: bar3
+      podSelector:
+        matchLabels:
+          foo2: bar2
+  ingress:
+  - action: Allow
+    enableLogging: false
+    from:
+    - namespaceSelector:
+        matchLabels:
+          foo3: bar3
+      podSelector:
+        matchLabels:
+          foo2: bar2
+    name: ""
+    ports:
+    - port: 80
+    to: null
+  priority: 10
+`,
+	}
+	Tier1 = test_resource{
+		inputResource: &crdv1alpha1.Tier{
+			TypeMeta: metav1.TypeMeta{Kind: "Tier", APIVersion: "crd.antrea.io/v1alpha1"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "TierA",
+			},
+			Spec: crdv1alpha1.TierSpec{
+				Priority:    10,
+				Description: "This is a test tier",
+			},
+		},
+		expPath: "/antrea-tiers/TierA.yaml",
+		expYaml: `apiVersion: crd.antrea.io/v1alpha1
+kind: Tier
+metadata:
+  name: TierA
+spec:
+  description: This is a test tier
+  priority: 10
+`,
+	}
+)
+
+func TestSetupRepo(t *testing.T) {
+	tests := []struct {
+		name           string
+		inputResources []test_resource
+	}{
+		{
+			name:           "empty-test",
+			inputResources: []test_resource{},
+		},
+		{
+			name:           "basic-test",
+			inputResources: []test_resource{Np1, Np2, Np3, Anp1, Acnp1},
+		},
+		{
+			name:           "empty-K8s-test",
+			inputResources: []test_resource{Anp1, Acnp1},
+		},
+		{
+			name:           "empty-CRDs-test",
+			inputResources: []test_resource{Np1, Np2},
+		},
+		{
+			name:           "tiers-test",
+			inputResources: []test_resource{Np1, Np2, Anp1, Tier1},
+		},
+	}
+	for _, test := range tests {
+		var expectedPaths = []string{}
+		var expectedYamls = []string{}
+		var resources = []runtime.Object{}
+		for _, resource := range test.inputResources {
+			resources = append(resources, resource.inputResource)
+			expectedPaths = append(expectedPaths, resource.expPath)
+			expectedYamls = append(expectedYamls, resource.expYaml)
+		}
+		fakeClient := NewClient(resources...)
+		k8s := &K8sClient{
+			Client: fakeClient,
+		}
+		runSetupTest(t, k8s, expectedPaths, expectedYamls)
+	}
+}
+
+func runSetupTest(t *testing.T, k8s *K8sClient, expPaths []string, expYamls []string) {
+	cr, err := SetupRepo(k8s, StorageModeInMemory, dir)
+	if err != nil {
+		t.Errorf("Error (TestSetupRepo): unable to set up repo")
+	}
```

Comment:
all these checks should use testify

```go
assert.NoError(t, err, "unable to set up repo")
```
or
```go
require.NoError(t, err, "unable to set up repo")
```
based on the rest of the test and the expected flow